docs/extension-development/check-if-woo-is-active.md

```diff
@@ -4,28 +4,45 @@ menu_title: Check if WooCommerce is active
 tags: how-to
 ---
 
-You can wrap your plugin in a check to see if WooCommerce is installed:
+When developing for WooCommerce, it's important to check that WooCommerce is installed and active before your own code runs. This ensures no errors occur from missing WooCommerce functions or classes.
+
+There are a few methods to achieve this. The most reliable approach is to initiate your code on `plugins_loaded`. This means you can be sure that WooCommerce has already initiated, by checking if one of its core classes exists.
 
 ```php
-// Test to see if WooCommerce is active (including network activated).
+add_action( 'plugins_loaded', 'prefix_plugins_loaded' );
 
-$plugin_path = trailingslashit( WP_PLUGIN_DIR ) . 'woocommerce/woocommerce.php';
+function prefix_plugins_loaded() {
+	// Check if the "WooCommerce" class exists. If not, exit early.
+	if ( ! class_exists( 'WooCommerce' ) ) {
+		return;
+	}
```

Comment:
This is essentially the same as hooking into `do_action( 'woocommerce_loaded' )` which is already executed on the `plugins_loaded` hook? Why not recommend it?

https://github.com/woocommerce/woocommerce/blob/f383284ba6dbef974be52d1fbf1e68f6517530fe/plugins/woocommerce/includes/class-woocommerce.php#L208

Reply:
Does `WooCommerce::on_plugins_loaded()` ever execute?

Reply:
@henrywright It's called via an action hook `add_action( 'plugins_loaded', array( $this, 'on_plugins_loaded' ), -1 );`